build/Build.cs

```diff
@@ -0,0 +1,137 @@
+
+[CheckBuildProjectConfigurations]
+[ShutdownDotNetAfterServerBuild]
+class Build : NukeBuild
+{
+    [Parameter("Configuration to build - Default is 'Debug' (local) or 'Release' (server)")]
+    readonly Configuration Configuration = IsLocalBuild ? Configuration.Debug : Configuration.Release;
+
+    [NukeOctoVersion] readonly OctoVersionInfo OctoVersionInfo;
+
+    [Solution(GenerateProjects = true)] readonly Solution Solution;
+
+    [Parameter("Test filter expression", Name = "where")] readonly string TestFilter = string.Empty;
+    [Parameter, Secret] readonly string FeedzIoApiKey;
+    [Parameter, Secret] readonly string NuGetApiKey;
+
+    AbsolutePath SourceDirectory => RootDirectory / "source";
+    AbsolutePath ArtifactsDirectory => RootDirectory / "artifacts";
+    AbsolutePath LocalPackagesDirectory => RootDirectory / ".." / "LocalPackages";
+
+    Target Clean => _ => _
+        .Before(Restore)
+        .Executes(() =>
+        {
+            SourceDirectory.GlobDirectories("**/bin", "**/obj", "./source/**/TestResults").ForEach(DeleteDirectory);
+            EnsureCleanDirectory(ArtifactsDirectory);
+        });
+
+    Target Restore => _ => _
+        .DependsOn(Clean)
+        .Executes(() =>
+        {
+            DotNetRestore(s => s
+                .SetProjectFile(Solution));
+        });
+
+    Target Compile => _ => _
+        .DependsOn(Restore)
+        .Executes(() =>
+        {
+            DotNetBuild(s => s
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .SetVersion(OctoVersionInfo.FullSemVer)
+                .SetAssemblyVersion(OctoVersionInfo.MajorMinorPatch)
+                .SetInformationalVersion(OctoVersionInfo.InformationalVersion)
+                .EnableNoRestore());
+        });
+
+    Target Test => _ => _
+        .DependsOn(Compile)
+        .Executes(() =>
+        {
+            DotNetTest(_ => _
+                .SetProjectFile(Solution)
+                .SetConfiguration(Configuration)
+                .SetLoggers("trx")
+                .SetVerbosity(DotNetVerbosity.Normal)
+                .SetFilter(TestFilter)
+                .EnableNoBuild()
+                .EnableNoRestore());
+            GlobFiles(SourceDirectory, "**/*.trx")
+                .ForEach(x => CopyFileToDirectory(x, ArtifactsDirectory, FileExistsPolicy.Overwrite));
+        });
+
+    Target Pack => _ => _
+        .DependsOn(Compile)
+        .DependsOn(Test)
+        .Executes(() =>
+        {
+            DotNetPack(_ => _
+                .SetProject(Solution)
+                .SetConfiguration(Configuration)
+                .SetOutputDirectory(ArtifactsDirectory)
+                .EnableNoBuild()
+                .AddProperty("Version", OctoVersionInfo.FullSemVer)
+            );
+        });
+
+    Target Publish => _ => _
+        .OnlyWhenStatic(() => IsServerBuild)
+        .DependsOn(Pack)
+        .TriggeredBy(Pack)
```

Comment:
Do you want this to be triggered by Pack too? This looks to be different behaviour to cake, seems like DependsOn might be all we want?

Reply:
This is because nuke doesn't run dependencies if the current target is skipped.
So if the `Default` target calls this one, then nothing will happen - need to invert it, like we do for `CopyToLocalPackages`

Reply:
I prefer to keep the flow one way, that way being from the end back to the start. Makes it easier to follow than using `TriggeredBy` and easier to follow than having two ways. 

Maybe you could have:

```
Target Default => _ => _
.DependsOn(Publish)
.DependsOn(CopyToLocalPackages);
```

That way, your pack target is always going to get called, and then it performs the desired outcome for local/server still, and keeps the flow one direction

Reply:
Sold.